File: brantham/linkedin/week-01-2026-04-28/render.py
import asyncio
from pathlib import Path
from playwright.async_api import async_playwright

HERE = Path(__file__).parent
FILES = [
    "post-01-top5-secteurs.html",
    "post-02-checklist-rj.html",
    "post-03-vrai-prix.html",
]

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        ctx = await browser.new_context(
            viewport={"width": 1080, "height": 1080},
            device_scale_factor=2,
        )
        page = await ctx.new_page()
        for f in FILES:
            url = f"file://{HERE / f}"
            await page.goto(url, wait_until="networkidle")
            await page.wait_for_timeout(800)
            card = page.locator(".post").first
            out = HERE / f.replace(".html", ".png")
            await card.screenshot(path=str(out), omit_background=False)
            print("rendered", out)
        await browser.close()

asyncio.run(main())

File: brantham/linkedin/week-01-2026-04-28/render.mjs
import { chromium } from 'playwright';
import { fileURLToPath } from 'url';
import path from 'path';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const files = [
  'post-01-top5-secteurs.html',
  'post-02-checklist-rj.html',
  'post-03-vrai-prix.html',
];

const browser = await chromium.launch();
const ctx = await browser.newContext({
  viewport: { width: 1080, height: 1080 },
  deviceScaleFactor: 2,
});
const page = await ctx.newPage();

for (const f of files) {
  const url = 'file://' + path.join(__dirname, f);
  await page.goto(url, { waitUntil: 'networkidle' });
  await page.waitForTimeout(800); // ensure web fonts settle
  const card = await page.locator('.post').first();
  const out = path.join(__dirname, f.replace('.html', '.png'));
  await card.screenshot({ path: out, omitBackground: false });
  console.log('rendered', out);
}

await browser.close();
